internal/repositories/db.go

```diff
@@ -0,0 +1,82 @@
+package repositories
+
+import (
+	"database/sql"
+	_ "github.com/jackc/pgx/v5"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"log"
+)
+
+type Link struct {
+	gorm.Model
+	ID          uint   `gorm:"primaryKey"`
+	OriginURL   string `gorm:"uniqueIndex"`
+	ShortLinkID string `gorm:"uniqueIndex"`
+}
+
+type DBRepo struct {
+	db *gorm.DB
+}
+
+func NewDBRepo(databaseDSN string) LinkRepo {
+	sqlDB, err := sql.Open("pgx", databaseDSN)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	db, err := gorm.Open(postgres.New(postgres.Config{
+		Conn: sqlDB,
+	}), &gorm.Config{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = db.AutoMigrate(&Link{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &DBRepo{
+		db: db,
+	}
+}
+
+func (r *DBRepo) GetOriginLink(shortLinkID string) (originLink string, ok bool) {
+	var link Link
+	r.db.First(&link, "short_link_id = ?", shortLinkID)
```

Comment:
Функция создает транзакцию и она не закрывается, так делать нельзя. 
```suggestion
	err:=r.db.First(&link, "short_link_id = ?", shortLinkID).Error
```
и вот эту ошибку уже обрабатывать. Ошибку работы с бд

Reply:
Функция не создает транзакцию и я обрабатываю ошибку ниже на две строчки

![2023-07-25 08 49 38](https://github.com/avidyakov/shortener/assets/89394067/4fa43a99-0f21-4616-9115-e324cc3802a4)